File: src/game/level/world2/World2Level1.java
package game.level.world2;

import game.entity.enemies.boss.Boss;
import game.entity.enemies.npcs.*;
import game.entity.models.Enemy;
import game.level.Level;
import game.level.world1.World1Level2;

public class World2Level1 extends World2Level {
    @Override
    public int getLevelId() {
        return 1;
    }
    @Override
    public int startEnemiesCount() {
        return 7;
    }

    @Override
    public Class<? extends Enemy>[] availableEnemies() {
        return new Class[]{
                Eagle.class,
                Zombie.class,
        };
    }

    @Override
    public Class<? extends Level> getNextLevel() {
        return World2Level2.class;
    }
}

File: src/game/level/world2/World2Level1.kt
package game.level.world2

import game.entity.enemies.npcs.Eagle
import game.entity.enemies.npcs.FastEnemy
import game.entity.models.Enemy
import game.level.Level

class World2Level1 : World2Level() {
    override val levelId: Int
        get() {
            return 1
        }

    override val startEnemiesCount: Int
        get() {
            return 7
        }

    override val availableEnemies: Array<Class<out Enemy>>
        get() {
            return arrayOf(
                    Eagle::class.java,
                    FastEnemy::class.java
            )
        }

    override val nextLevel: Class<out Level?>
        get() {
            return World2Level2::class.java
        }
}